components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -85,6 +85,11 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
             if (LOG.isDebugEnabled()) {
                 LOG.debug("Client Error while sending recovery notification ", e);
             }
+            if (StringUtils.equals(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_FEDERATED_USER.getCode(),
+                    e.getErrorCode()) && StringUtils.equals(String.format(IdentityRecoveryConstants.ErrorMessages
```

Comment:
Why we need to check the error message equality in here?

Reply:
The same error code (18004) has beed used for two error messages .
1. https://github.com/wso2-extensions/identity-governance/blob/ccfe40e7ddd05c239638ab726336bca7389e0701/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java#L203
2. https://github.com/wso2-extensions/identity-governance/blob/ccfe40e7ddd05c239638ab726336bca7389e0701/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java#L290

Reply:
Removed the error message check by changing the error code to a unique one.